ACM_General/home/views.py

```diff
@@ -1,16 +1,23 @@
 from django.shortcuts import render
-
+from events.models import Event
+from django.utils import timezone
+from django.conf import settings
 
 # Create your views here.
 
 
 def index(request):
-    return (
-        render(
-            request,
-            'home/index.html',
-        )
-    )
+
+    # Grabs the 3 nearest upcoming events that haven't expired yet
+    events = Event.objects.all().order_by('date_hosted').filter(date_expire__gte=timezone.now())
```

Comment:
Is there any way to do this optimally in the events so it doesn't query all events and then does `len()`? This may prove resource intensive as more events are created. By optimally I mean instead of `all()`.